gatsby-node.js

```diff
@@ -50,6 +48,20 @@ exports.createPages = ({ graphql, actions }) => {
         // Create blog posts pages.
         const posts = result.data.allMarkdownRemark.edges;
 
+        const _originalTitleCache = new Map();
+        const originalTitle = slug => {
+          const plainSlug = slug.replace(/^\/[\w-]+\//, '/');
+          if (_originalTitleCache.has(plainSlug)) {
+            return _originalTitleCache.get(plainSlug);
+          }
+          const post = posts.find(({ node }) => node.fields.slug === plainSlug);
+          if (!post)
+            throw new Error('Can’t find parent for post with slug ' + slug);
+          const { title } = post.node.frontmatter;
+          _originalTitleCache.set(plainSlug, title);
+          return title;
```

Comment:
Something like this could actually be useful for https://github.com/gaearon/overreacted.io/issues/99